apps/gp2-server/src/routes/public/working-group.route.ts

```diff
@@ -0,0 +1,49 @@
+import { Router, Response } from 'express';
+import { gp2 as gp2Model } from '@asap-hub/model';
+import WorkingGroupController from '../../controllers/working-group.controller';
+
+export const workingGroupRouteFactory = (
+  workingGroupController: WorkingGroupController,
+): Router => {
+  const workingGroupRoutes = Router();
+
+  workingGroupRoutes.get(
+    '/working-groups',
+    async (_req, res: Response<gp2Model.ListPublicWorkingGroupResponse>) => {
+      const result = await workingGroupController.fetch();
+
+      res.json({
+        total: result.total,
+        items: result.items.map(mapWorkingGroupToPublicWorkingGroup),
+      });
+    },
+  );
+
+  workingGroupRoutes.get(
+    '/working-groups/:workingGroupId',
+    async (req, res: Response<gp2Model.PublicWorkingGroupResponse>) => {
+      const { workingGroupId } = req.params;
+
+      const workingGroup =
+        await workingGroupController.fetchById(workingGroupId);
+
+      res.json(mapWorkingGroupToPublicWorkingGroup(workingGroup));
+    },
+  );
+
+  return workingGroupRoutes;
+};
+
+const mapWorkingGroupToPublicWorkingGroup = (
+  workingGroup: gp2Model.WorkingGroupResponse,
+): gp2Model.PublicWorkingGroupResponse => ({
+  id: workingGroup.id,
+  description: workingGroup.description,
+  members: workingGroup.members,
```

Comment:
Hey, I think we need more data on members, don't we?

It's listed in ticket

- degree
- role
- dept
- institution
- city
- country

and in the website (https://gp2.org/working-group-directory/) passed as reference we have

![Screenshot 2024-05-09 at 07 18 43](https://github.com/yldio/asap-hub/assets/16595804/d183aeaf-9dbe-46b1-9a57-5ac733202a1e)

so these info seems to be used

Reply:
as per our discussion - we decided to leave these out and GMMB don't seem to need any extra information embedded for the nested resources